src/app/features/profile/pages/change-password/change-password.component.ts

```diff
@@ -0,0 +1,66 @@
+import {Component, OnInit} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from "@angular/forms";
+import {ProfileService} from "../../service/profile.service";
+import {Router} from "@angular/router";
+import {Profile} from "../../models/profile";
+import {Observable} from "rxjs";
+import {NotificationService} from "../../../../core/services/notification.service";
+import {MenuItem} from "primeng/api";
+
+@Component({
+  selector: 'app-change-password',
+  templateUrl: './change-password.component.html',
+  styleUrl: './change-password.component.scss'
+})
+export class ChangePasswordComponent implements OnInit{
+  passwordForm: FormGroup;
+  menuItems: MenuItem[] = [
+    { label: 'Profile', icon: 'pi pi-user', routerLink: ['/profile'] },
+    { label: 'change personnal informations', icon: 'pi pi-key', routerLink: ['/profile/update'] },
+  ];
+  profile$!: Observable<Profile>;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private profileService: ProfileService,
+    private notificationService: NotificationService,
+    private router: Router
+  ) {
+    this.passwordForm = this.formBuilder.group({
+      oldPassword: ['', Validators.required],
+      newPassword: ['', [Validators.required, Validators.minLength(8)]]
+    });
+  }
+  ngOnInit(): void {
+    this.profile$ = this.profileService.getProfile();
+  }
+
+  onSubmit() {
+    if (this.passwordForm.valid) {
+      const request = {
+        oldPassword: this.passwordForm.value.oldPassword,
+        newPassword: this.passwordForm.value.newPassword
+      };
+
+      this.profileService.userChangePassword(request).subscribe(
```

Comment:
Instead of creating a creating a request object here, can't you just send the formGroup like this ? : 
```
this.profileService.userChangePassword(this.passwordForm.value).subscribe(...)
```